File: frontend/src/Types/Recipe.ts
import type {User} from "./User.ts";

export interface Recipe {
    id:number,
    recipeTitle:string,
    description?: string,
    ingredients: string,
    instructions: string,
    category?: "DESSERT" | "SOUP" | "MAIN" | "DRINK",
    pictureSrc?: string,
    user?:User,
    username:string
}

export type recipeType =  {
    id:number,
    recipeTitle:string,
    description?: string,
    ingredients: string,
    instructions: string,
    category?: "DESSERT" | "SOUP" | "MAIN" | "DRINK",
    pictureSrc?: string,
    user?:User,
    username:string
}

File: backend/src/main/java/com/example/demo/model/Recipe.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "recipes")
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String recipeTitle;

    @Column(nullable = false)
    private String description;

    @Column(length = 200000)
    private String ingredients;

    @Column(length = 200000)
    private String instructions;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Category category;

    @Column(nullable = false)
    private String pictureSrc;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;


}
